### PR TITLE
ROX-11939: Enable Scanner monitoring exposure

### PR DESCRIPTION
**Pull request by stehessel**
Recently we have added an option for the ACS operator to enable monitoring endpoint for Scanner in https://github.com/stackrox/stackrox/pull/2748.

With this PR, we are using a new option and enabling monitoring during provisioning of ACS operator by Fleetshard Synchronizer.

Changes

- added flag to enable exposing of monitoring port for Scanner.
- updated stackrox "lib" to have a new Scanner struct.
- fixed observatorium mock signatures, which was required due to the version update.

Note that older versions of the ACS operator (before https://github.com/stackrox/stackrox/pull/2748) will simply ignore the monitoring flag. It will not exist in YAML of the Scanner custom resource.

See also https://github.com/stackrox/acs-fleet-manager/pull/158, which adds the equivalent monitoring flag for Central.

## Checklist (Definition of Done)

- [ ] ~Unit and integration tests added~
- [ ] ~Documentation added if necessary~
- [ ] CI and all relevant tests are passing
- [x] Discussed security and business related topics privately. Will move any security and business related topics that arise to private communication channel.

## Test manual

Tested Scanner metrics scrape on dev data plane cluster using a custom RHACS operator `3.71.0-378-g45479f2f35`.

### CODE REVIEW

File: pkg/client/observatorium/api_mock.go [hunk 1]
```diff
@@ -15,17 +15,16 @@ func (c *Client) MockAPI() pV1.API {
 	return &httpAPIMock{}
 }
 
-type httpAPIMock struct {
-}
+type httpAPIMock struct{}
 
 // Query performs a query for the dinosaur metrics.
-func (t *httpAPIMock) Query(ctx context.Context, query string, ts time.Time) (pModel.Value, pV1.Warnings, error) {
+func (t *httpAPIMock) Query(ctx context.Context, query string, ts time.Time, opts ...pV1.Option) (pModel.Value, pV1.Warnings, error) {
```
Comment (mtesseract) [inline pkg/client/observatorium/api_mock.go:21]: Is this an unrelated change?

**Review by mtesseract — APPROVED** [review]

File: pkg/client/observatorium/api_mock.go (hunk 1 above)
Comment (kovayur) [reply pkg/client/observatorium/api_mock.go:21]: There is a change in API. Attempted also to fix in #330

Comment (stehessel) [reply pkg/client/observatorium/api_mock.go:21]: This change got pulled it from the dependency update, which I did to update the Scanner CRD in `stackrox/rox` to latest version.

Comment (stehessel) [reply pkg/client/observatorium/api_mock.go:21]: @kovayur Your fix seems much more rigorous than mine. Do you prefer that I rebase this PR to #330 or should I just merge and you take it from there?

Comment (kovayur) [reply pkg/client/observatorium/api_mock.go:21]: @stehessel feel free to merge your PR, I'll rebase then.
